### DISCUSSION TITLE
Generate aggregate project with the rest of config files

### DISCUSSION THREAD

**Discussion by mosteo** [alire-project/alire#801] [Ideas]
@Fabien-Chouteau , I remember you've said sometimes that aggregate projects have shortcomings, so we are staying away from them to drive builds and so on.

I wondered though if a basic usage similar to our `alr_env.gpr` file, just to set the environment, could be useful for example to allow users to open it with GNAT Studio. This would allow adding dependencies and reloading the project without leaving the editor.

We could generate it with the rest of the config files with the same environment that `alr printenv` displays. I guess this should work for most regular projects?

**Comment by Fabien-Chouteau** [comment]
I think they don't mix well for cross projects.
And I don't think we will be able to set environment variables like `PATH`.

**Comment by mosteo** [comment]
Right, I always forget about cross projects.

About the PATH, is that really necessary for edition only? (That's the only use case I have in mind.)

**Reply by Fabien-Chouteau** [reply]
> About the PATH, is that really necessary for edition only? (That's the only use case I have in mind.)

Yes because that's what will give you the toolchain now.
